File: tests/fixtures/scala/scope.scala
package fixtures.scope

import vendor.Worker as ImportedWorker

object Scope:
  val service = createService()

  def run(loader: () => String): Unit =
    loader()
    helper()
    service.execute()
    ImportedWorker.create()

    val callback = () => perform()
    callback()
    new LocalWorker(build())

  def helper(): Unit = ()

class LocalWorker(value: String)

File: tests/fixtures/csharp/scope.cs
using WorkerAlias = Fixtures.Worker;

class Scope
{
    private readonly Service service = CreateService();

    void Run(Func<string> loader)
    {
        loader();
        Helper();
        service.Execute();
        WorkerAlias.Create();

        Func<string, string> normalize = value => value.Trim();
        normalize(" value ");

        string Local(string value) { return Clean(value); }
        Local("x");

        Worker worker = new Worker(Build());
        Worker other = new();
        worker?.Execute();
        worker!.Execute();
    }

    void Helper() { }
}

class Worker
{
    public static Worker Create() => new Worker();
    public void Execute() { }
}
